scipy/stats/_stats.pyx

```diff
@@ -753,3 +753,90 @@ def gaussian_kernel_estimate(points, values, xi, precision, dtype, real _=0):
                 estimate[j, k] += values_[i, k] * arg
 
     return np.asarray(estimate)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def gaussian_kernel_estimate_log(points, values, xi, precision, dtype, real _=0):
+    """
+    def gaussian_kernel_estimate_log(points, real[:, :] values, xi, precision)
+
+    Evaluate the log of the estimated pdf on a provided set of points.
+
+    Parameters
+    ----------
+    points : array_like with shape (n, d)
+        Data points to estimate from in ``d`` dimensions.
+    values : real[:, :] with shape (n, p)
+        Multivariate values associated with the data points.
+    xi : array_like with shape (m, d)
+        Coordinates to evaluate the estimate at in ``d`` dimensions.
+    precision : array_like with shape (d, d)
+        Precision matrix for the Gaussian kernel.
+
+    Returns
+    -------
+    estimate : double[:, :] with shape (m, p)
+        The log of the multivariate Gaussian kernel estimate evaluated at the
+        input coordinates.
+    """
+    cdef:
+        real[:, :] points_, xi_, values_, log_values_
+        real[:, :] whitening, estimate, estimate_, zeros_estimate_
+        real[:] max_, zeros_max
+        int i, j, k
+        int n, d, m, p
+        real arg, residual, log_arg, log_norm
+
+    n = points.shape[0]
+    d = points.shape[1]
+    m = xi.shape[0]
+    p = values.shape[1]
+
+    # Rescale the data
+    whitening = np.linalg.cholesky(precision).astype(dtype, copy=False)
+    points_ = np.dot(points, whitening).astype(dtype, copy=False)
+    xi_ = np.dot(xi, whitening).astype(dtype, copy=False)
```

Comment:
If we want `np.dot(x, whitening)` but have only the covariance matrix `A` instead of its inverse, `precision`, we can do:
```python3
JAJ = A[::-1, ::-1]
L2 = linalg.cholesky(JAJ)
JL2J = L2[::-1, ::-1]
res2 = linalg.solve_triangular(JL2J, x.T, lower=True).T
```
I expect this will more numerically favorable as it avoids the matrix inversion, and although all the indexing/transposition might mean that copies are needed before LAPACK is called, it should be a bit faster otherwise because: 
* an inversion is replaced with Cholesky decomposition
* a cholesky decomposition is replaced with a triangular solve
* a matrix-matrix product is eliminated

Example:
<details>

```python3
import numpy as np
from scipy import linalg

np.random.seed(0)

n = 10
x = np.random.rand(n, n)
A12 = np.random.rand(n, n)
A = A12 @ A12.T

P = linalg.inv(A)
Lp = linalg.cholesky(P)
res = np.dot(x, Lp)

JAJ = A[::-1, ::-1]
L2 = linalg.cholesky(JAJ)
JL2J = L2[::-1, ::-1]
res2 = linalg.solve_triangular(JL2J, x.T, lower=True).T

np.testing.assert_allclose(res, res2)
```

</details>

Reply:
gh-16692 implements this idea for `pdf` and both benchmarks execute in half the time as main.